### ISSUE TITLE
No symbols have been loaded. Visual Studio 2022

### ISSUE THREAD

**Issue by AndrewWBurnett** [fmtlib/fmt#4157]
Hi,

I am having an odd issue with fmt, CMake and Visual Studio. From what I can work out, it is something the fmt CMakeLists.txt file is doing which is causing my issue. I have probably missed something in the documentation, but thought I'd file a report with you guys in case it was an error in the fmt CMakeLists.txt file.

Simply put, I am finding it difficult to get a Debug build's breakpoints working in Visual Studio 2022 when the project is built using CMake. 

By this I mean, I write a CMake file which fetches the latest version of fmt and then builds a very simple executable. When I then open the created solution in Visual Studio, insert a breakpoint into the code and run the code, no breakpoints are hit and I am told "No symbols have been loaded for the document".

Versions of software:
- Visual Studio version 17.11.2
- CMake version: 3.27.7
- fmt version: master

I have included a minimal example:
CMakeLists.txt
```
cmake_minimum_required(VERSION 3.23)
#set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /Zi")
include(FetchContent)
FetchContent_Declare(
  fmt
  GIT_REPOSITORY 	https://github.com/fmtlib/fmt.git
  GIT_TAG 			master
)
FetchContent_MakeAvailable(fmt)
project(fmt_test)
add_executable(fmt_test_run main.cpp)
#target_link_libraries(fmt_test_run PUBLIC fmt)
#add_dependencies(fmt_test_run fmt)
```

main.cpp
```
#include <iostream>
int main() {
    for (int i{ 0 }; i < std::numeric_limits<int>::max(); ++i)
    {
        std::cout << i << std::endl;
    }
    return 0;
}
```
I am inserting the breakpoint on the line with `std::cout'.

My build tree is very simple; main.cpp and CMakeLists.txt are together in a folder. I create a build directory in that folder, navigate to it, then run `build..`.
 

I know I have not linked my executable against fmt, however I do not need for this behaviour to occour. The line that causes this odd behaviour is `FetchContent_MakeAvailable(fmt)`. If you comment that out an re-build, you will find that the breakpoints are now working correctly.

Now, I can fix this by uncommenting out the line `set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /Zi")`. Then, everything works fine. However, I would prefer not to have to use this if possible.

I think this is something to do with the pdb files, however I am really not an expert in them or in CMake. I know fmt does some things using the pdb file which are not usual, however I don't have the expertise to say if that is causing the issue I'm having. I thought I'd ask and see if there is either a solution, a workaround, or someone can come up with a fix. Or, I have missed something in the documentation, and this is not an issue. I did have a look online (and in other issues in the tracker) but could not find anything regarding the problems I am having.

Thank you

**Comment by vitaut** [comment]
This looks more like a CMake than {fmt} question, you might want to ask in CMake forum or StackOverflow. FWIW {fmt} tries to avoid modifying the global state and works with target properties whenever possible.

**Comment by AndrewWBT** [comment]
Hi,

So after investigating this, I found the two fixes for this issue. In short, it was CMake's issue, not fmt's. However I thought I would post, in case anyone else got in the same situation I did and came here looking for a resolution.

The issue is with the cmake_minimum_required code, but not on all versions of CMake. For some reason, when fmt sets cmake_minimum_required in the way that it does, when used with my code using 3.23, this behaviour occours. I do not know why, I did not investigate the why.

The issue can be solved in one of two ways:
1) Use a more recent version of CMake than I did. Specifically, by changing the line `cmake_minimum_required(VERSION 3.23)` to `cmake_minimum_required(VERSION 3.30)`, this issue resolves itself. I am unsure of on which version this issue was resolved, whether it was just 3.23, or if there are a range of versions this issue presents itself in.
2) Change the line `cmake_minimum_required(VERSION 3.8...3.28)` in fmt's CMakeLists.txt to something else I found that using  `cmake_minimum_required(VERSION 3.23)` resolved my issues.

Hopefully this helps someone!
